lib/std/debug.zig

```diff
@@ -1527,9 +1527,9 @@ pub fn ConfigurableTrace(comptime size: usize, comptime stack_frame_count: usize
 }
 
 pub const SafetyLock = struct {
-    state: State = .unlocked,
+    state: State = if (runtime_safety) .unlocked else .unknown,
```

Comment:
It's a breaking change, but an alternate approach would be to remove the default value for this field, and then add a `default` decl:

```zig
    state: State,
    pub const State = if (runtime_safety) enum { unlocked, locked } else enum { unknown };
    pub const default: SafetyLock = if (runtime_safety) .{ .state = .unlocked } else .{ .state = .unknown };
```

Then in hash_map.zig you'd switch it to using a decl literal: `self.pointer_stability = .default;` (and there are a few other places in hash_map.zig and array_hash_map.zig that would have to be updated to the new syntax